### DISCUSSION TITLE
How to create custom oxlint plugins and rules?

### DISCUSSION THREAD

**Discussion by sagargurtu** [oxc-project/oxc#6275] [Q&A]
Similar to ESLint, is it possible to create custom plugins and rules for Oxlint? Is there any documentation available for that?

**Comment by eklmv** [comment]
Would love to know answer for those questions, and votes looks like i'm not alone. :eyes:

**Comment by nrayburn-tech** [comment]
It’s not possible at this time. You probably want to subscribe to https://github.com/oxc-project/oxc/discussions/10342.

Once custom plugins are available, there will be documentation to go with it.

**Reply by eklmv** [reply]
hm, so it's not possible even in rust right now?

**Reply by nrayburn-tech** [reply]
No and I don’t think there’s plans to make it possible in Rust.

**Reply by eklmv** [reply]
thanks for your clarification
